lib/config_parser/parser.go

```diff
@@ -0,0 +1,78 @@
+package config_parser
+
+import (
+	sc "github.com/chef/automate/api/config/deployment"
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/chef/automate/lib/io/fileutils"
+	ptoml "github.com/pelletier/go-toml"
+)
+
+type ConfigParser interface {
+	ParseAWSAutomateConfig(configFile string) (*HAAwsConfigToml, error)
+	ParseOnPremConfig(configFile string) (*HAOnPremConfigToml, error)
+	ParseStandaloneConfig(configFile string) (*sc.AutomateConfig, error)
+}
+
+type ConfigParserImpl struct{}
+
+func (cp *ConfigParserImpl) ParseAWSAutomateConfig(configFile string) (*HAAwsConfigToml, error) {
+	return ParseAWSAutomateConfig(configFile)
+}
+
+func (cp *ConfigParserImpl) ParseOnPremConfig(configFile string) (*HAOnPremConfigToml, error) {
+	return ParseOnPremConfig(configFile)
+}
+
+func (cp *ConfigParserImpl) ParseStandaloneConfig(configFile string) (*sc.AutomateConfig, error) {
+	return ParseStandaloneConfig(configFile)
+}
+
+func ParseAWSAutomateConfig(configFile string) (*HAAwsConfigToml, error) {
+
+	/* This function will read the config toml file and will try to parse it in the structure.
```

Comment:
if the logic of two functions are same then write a common function and use it both